src/components/PostDetails.tsx

```diff
@@ -1,106 +1,124 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { createComment, deleteComment, getPostComments } from './api/users';
+import { Comment } from '../types/Comment';
+
+type Props = {
+  selectedPost: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ selectedPost }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isErrorComments, setIsErrorComments] = useState(false);
+  const [isShownForm, setIsShownForm] = useState(false);
+
+  const isNoComments =
+    selectedPost && comments.length === 0 && !isLoading && !isErrorComments;
+
+  useEffect(() => {
+    setIsErrorComments(false);
+    setIsShownForm(false);
+    if (selectedPost) {
+      setIsLoading(true);
+      getPostComments(selectedPost.id)
+        .then(setComments)
+        .catch(() => setIsErrorComments(true))
+        .finally(() => setIsLoading(false));
+    }
+  }, [selectedPost]);
+
+  const addComment = ({ name, email, body, postId }: Omit<Comment, 'id'>) => {
```

Comment:
It's better to validate the input data before creating a comment. Make sure you can't add a comment with empty data (with spaces only). This is a requirement from the checklist.